pkg/install/vault/init.go

```diff
@@ -0,0 +1,212 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+
+	vaultapi "github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+	"github.com/puppetlabs/leg/vaultutil/pkg/model"
+	vaultutil "github.com/puppetlabs/leg/vaultutil/pkg/vault"
+)
+
+type VaultInitializer struct {
+	vaultConfig        *vaultutil.VaultConfig
+	vaultCoreConfig    *VaultCoreConfig
+	vaultSystemManager model.VaultSystemManager
+}
+
+func (vi *VaultInitializer) InitializeVault(ctx context.Context) error {
+	credentials := &model.VaultKeys{
+		RootToken:  vi.vaultConfig.Token,
+		UnsealKeys: []string{vi.vaultConfig.UnsealKey},
+	}
+
+	var err error
+	if credentials.RootToken == "" || credentials.UnsealKeys[0] == "" {
+		credentials, err = vi.vaultSystemManager.GetCredentials(ctx)
+		if err != nil {
+			return err
+		}
+
+		if credentials == nil {
+			credentials, err = vi.vaultSystemManager.Initialize(ctx)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	err = vi.vaultSystemManager.Unseal(credentials)
+	if err != nil {
+		return err
+	}
+
+	vi.vaultSystemManager.SetToken(credentials)
+
+	err = vi.vaultSystemManager.EnableKubernetesAuth()
+	if err != nil {
+		return err
+	}
+
+	err = vi.vaultSystemManager.ConfigureKubernetesAuth(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = vi.vaultSystemManager.EnableJWTAuth()
+	if err != nil {
+		return err
+	}
+
+	err = vi.vaultSystemManager.ConfigureJWTAuth(ctx)
+	if err != nil {
+		return err
+	}
+
+	// FIXME Upgrade to the latest oauthapp plugin
+	plugins := []*vaultapi.RegisterPluginInput{
```

Comment:
You actually don't need this for relay-core, just the API, so should be safe to remove here (for now, anyway).

Reply:
Cool. The `relay` dev environment was likely doing this preemptively on behalf of the API. I'll adjust this in the API handling/workflow as necessary instead.